uint16[8] channels_in
uint16[8] channels_out
bool lock